File: app/src/main/java/peter/util/searcher/activity/BaseActivity.java
package peter.util.searcher.activity;

import android.app.Activity;
import android.app.AlertDialog;
import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.text.format.DateUtils;
import android.util.Log;
import android.webkit.CookieManager;
import android.webkit.CookieSyncManager;

import java.io.File;
import java.util.ArrayList;
import java.util.Date;

import peter.util.searcher.R;
import peter.util.searcher.bean.TabBean;

/**
 * 基础activity 提供一些公用方法
 * Created by peter on 16/5/9.
 */
public class BaseActivity extends AppCompatActivity {

    public static final String ACTION_INNER_BROWSE = "peter.util.searcher.inner";
    public static final String NAME_BEAN = "peter.util.searcher.bean";
    public static final String NAME_WORD = "peter.util.searcher.word";
    private static final ArrayList<Activity> LIST = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LIST.add(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        LIST.remove(this);
    }

    public void exit() {
        for(Activity activity : LIST) {
            activity.finish();
        }
    }

    public String getSearchWord() {
        return "";
    }

    public void setSearchWord(String word) {
    }

    public void startBrowser(TabBean bean) {
        Intent intent = new Intent(BaseActivity.this, MainActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
        intent.setAction(ACTION_INNER_BROWSE);
        intent.putExtra(NAME_BEAN, bean);
        startActivity(intent);
    }

    public int clearCacheFolder(final File dir, final int numDays) {
        int deletedFiles = 0;
        if (dir != null && dir.isDirectory()) {
            try {
                for (File child : dir.listFiles()) {

                    //first delete subdirectories recursively
                    if (child.isDirectory()) {
                        deletedFiles += clearCacheFolder(child, numDays);
                    }

                    //then delete the files and subdirectories in this dir
                    //only empty directories can be deleted, so sub dirs have been done first
                    if (child.lastModified() < new Date().getTime() - numDays * DateUtils.DAY_IN_MILLIS) {
                        if (child.delete()) {
                            deletedFiles++;
                        }
                    }
                }
            } catch (Exception e) {
                Log.e("peter", String.format("Failed to clean the cache, error %s", e.getMessage()));
            }
        }
        return deletedFiles;
    }

    @SuppressWarnings("deprecation")
    public void ClearCookies() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1) {
            CookieManager.getInstance().removeAllCookies(null);
            CookieManager.getInstance().flush();
        } else {
            CookieSyncManager cookieSyncMng = CookieSyncManager.createInstance(BaseActivity.this);
            cookieSyncMng.startSync();
            CookieManager cookieManager = CookieManager.getInstance();
            cookieManager.removeAllCookie();
            cookieManager.removeSessionCookie();
            cookieSyncMng.stopSync();
            cookieSyncMng.sync();
        }
    }

    public void sendMailByIntent() {
        Intent data = new Intent(Intent.ACTION_SENDTO);
        data.setData(Uri.parse(getString(R.string.setting_feedback_address)));
        data.putExtra(Intent.EXTRA_SUBJECT, getString(R.string.setting_feedback));
        data.putExtra(Intent.EXTRA_TEXT, getString(R.string.setting_feedback_body));
        startActivity(data);
    }

    public void showAlertDialog(String title, String content) {
        AlertDialog dialog = new AlertDialog.Builder(BaseActivity.this).create();
        dialog.setCanceledOnTouchOutside(true);
        dialog.setTitle(title);
        dialog.setMessage(content);
        dialog.show();
    }

    public void showAlertDialog(int titleRes, int contentRes) {
        showAlertDialog(getString(titleRes), getString(contentRes));
    }

    public String getVersionName() {
        PackageManager packageManager = getPackageManager();
        PackageInfo packInfo = null;
        try {
            packInfo = packageManager.getPackageInfo(getPackageName(), 0);
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }

        String version = "";
        if (packInfo != null) {
            version = packInfo.versionName;
        }

        if (TextUtils.isEmpty(version)) {
            return "";
        } else {
            return "version " + version;
        }
    }

    public int getVersionCode() {//获取版本号(内部识别号)
        try {
            PackageInfo pi = getPackageManager().getPackageInfo(getPackageName(), 0);
            return pi.versionCode;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
            return 0;
        }
    }

}

File: app/src/main/java/peter/util/searcher/activity/BaseActivity.kt
package peter.util.searcher.activity

import android.app.Activity
import android.app.AlertDialog
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.format.DateUtils
import android.util.Log
import android.view.inputmethod.InputMethodManager
import android.webkit.CookieManager
import java.io.File
import java.util.ArrayList
import java.util.Date
import peter.util.searcher.R
import peter.util.searcher.bean.Bean

/**
 * 基础activity 提供一些公用方法
 * Created by peter on 16/5/9.
 */
open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        LIST.add(this)
    }

    open fun getSearchWord() : String? = null

    open fun setSearchWord(word : String) {}

    fun getVersionName() : String = packageManager.getPackageInfo(packageName, 0).versionName

    fun getVersionCode() : Int = packageManager.getPackageInfo(packageName, 0).versionCode

    override fun onDestroy() {
        super.onDestroy()
        LIST.remove(this)
    }

    fun exit() {
        for (activity in LIST) {
            activity.finish()
        }
    }

    fun closeIME() {
        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(currentFocus?.windowToken, 0)
    }

    fun openIME() {
        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.toggleSoftInput(InputMethodManager.SHOW_IMPLICIT, 0)
    }

    fun startBrowser(bean: Bean) {
        val intent = Intent(this, MainActivity::class.java)
        intent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
        intent.action = ACTION_INNER_BROWSE
        intent.putExtra(NAME_BEAN, bean)
        startActivity(intent)
    }

    fun clearCacheFolder(dir: File?, numDays: Int): Int {
        var deletedFiles = 0
        if (dir != null && dir.isDirectory) {
            try {
                for (child in dir.listFiles()) {

                    //first delete subdirectories recursively
                    if (child.isDirectory) {
                        deletedFiles += clearCacheFolder(child, numDays)
                    }

                    //then delete the files and subdirectories in this dir
                    //only empty directories can be deleted, so subdirs have been done first
                    if (child.lastModified() < Date().time - numDays * DateUtils.DAY_IN_MILLIS) {
                        if (child.delete()) {
                            deletedFiles++
                        }
                    }
                }
            } catch (e: Exception) {
                Log.e("peter", "Failed to clean the cache, error ${e.message}")
            }
        }
        return deletedFiles
    }

    fun clearCookies() {
        CookieManager.getInstance().removeAllCookies(null)
        CookieManager.getInstance().flush()
    }

    fun sendMailByIntent() {
        val data = Intent(Intent.ACTION_SENDTO)
        data.data = Uri.parse(getString(R.string.setting_feedback_address))
        data.putExtra(Intent.EXTRA_SUBJECT, getString(R.string.setting_feedback))
        data.putExtra(Intent.EXTRA_TEXT, getString(R.string.setting_feedback_body))
        startActivity(data)
    }

    fun showAlertDialog(titleRes: Int, contentRes: Int) {
        val dialog = AlertDialog.Builder(this@BaseActivity).create()
        dialog.setCanceledOnTouchOutside(true)
        dialog.setTitle(getString(titleRes))
        dialog.setMessage(getString(contentRes))
        dialog.show()
    }

    companion object {
        val ACTION_INNER_BROWSE = "peter.util.searcher.inner"
        val NAME_BEAN = "peter.util.searcher.bean"
        val NAME_WORD = "peter.util.searcher.word"
        private val LIST = ArrayList<Activity>()
    }

}
